docs/rules/saga-direct-delegation.md

```diff
@@ -0,0 +1,58 @@
+# Enforce using `yield*` to delegate to other sagas
+# saga-direct-delegation
+
+This rule enforces that delegation to sagas is done directly using `yield*`, rather than indirectly by plain `yield` of the returned iterator to the saga middleware.
+
+This has the benefit of maintaining the delegator (caller) on the call stack, which aids debugging and gives better error stack traces.
+
+The rule is autofixable.

```

Comment:
Autofixing existing code may cause/reveal issues in unit tests. Found the following so far

1. Mocking a nested saga using a function instead of a generator (revealed test implementation issue).

1. When testing iterators by driving them manually using `next()`, it's no longer possible to "step over" nested iterators returned (by doing nothing with them) and passing a mock value (if relevant) to the following `next()` call. Must drive it through, or mock it out (e.g. using `babel-plugin-rewire`). Latter preferable as it avoids needing to care about the nested generator. Neither as convenient or as elegant as what you can do with plain `yield`>

- [ ] Find any remaining sensitivities or implications for unit tests
- [ ] Add them in a "when not to use it" section

Reply:
063b230

(both cases are caused issues; 1. is not a revealed issue)